### PR TITLE
BROKER: Get Current Date Time

### PR DESCRIPTION
**Pull request by MahalakshmiMurari**
BROKER: Get Current Date Time
CODERUB: Infrastructure Build

### CODE REVIEW

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]
Update your yml too

**Comment by MahalakshmiMurari** [comment]
CODERUB: Ran Infrastructure build

File: ReserveSpot.Services.Api/Startup.cs [hunk 1]
```diff
@@ -21,6 +22,7 @@ public class Startup
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
+            services.AddSingleton<IDateTimeBroker, DateTimeBroker>();
```
Comment (ShriHumrudha) [inline ReserveSpot.Services.Api/Startup.cs]: Why Singleton?

Comment (ShriHumrudha) [reply ReserveSpot.Services.Api/Startup.cs]: Ping

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ReserveSpot.Services.Api/Startup.cs (hunk 1 above)
Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Startup.cs]: Date and Time can be created and set for the first time and can be used further based on the request .
As Singleton lifetime services are created the first time they are requested and then every subsequent request will use .. so we used singleton.

**Review by ShriHumrudha — APPROVED** [review]

File: ReserveSpot.Services.Api/Startup.cs (hunk 1 above)
Comment (ShriHumrudha) [reply ReserveSpot.Services.Api/Startup.cs]: Stay careful with Singleton. If there is no real need to keep the state consistent then you can prefer Transient.

Comment (MahalakshmiMurari) [reply ReserveSpot.Services.Api/Startup.cs]: Changed the services to Transient. Please review and approve the PR

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**